install/management/dispatcher.go

```diff
@@ -0,0 +1,250 @@
+	"github.com/vmware/vic/install/configuration"
+	"github.com/vmware/vic/pkg/errors"
+	"github.com/vmware/vic/pkg/vsphere/compute"
+	"github.com/vmware/vic/pkg/vsphere/diagnostic"
+	"github.com/vmware/vic/pkg/vsphere/session"
+	"github.com/vmware/vic/pkg/vsphere/tasks"
+	"github.com/vmware/vic/pkg/vsphere/vm"
+
+	"golang.org/x/net/context"
+)
+
+var (
+	insecure = true
+)
+
+type Dispatcher struct {
+	session *session.Session
+	ctx     context.Context
+	force   bool
+
+	isVC          bool
+	vchPoolPath   string
+	vmPathName    string
+	dockertlsargs string
+	DockerPort    string
+	HostIP        string
+
+	vchPool   *compute.ResourcePool
+	appliance *vm.VirtualMachine
+
+	// TODO: remove temp structure after refactor network configuration
+	networks map[string]object.NetworkReference
+	nics     map[string]string
+}
+
+type diagnosticLog struct {
+	key     string
+	name    string
+	start   int32
+	host    *object.HostSystem
+	collect bool
+}
+
+var diagnosticLogs = make(map[string]*diagnosticLog)
+
+func NewDispatcher(conf *configuration.Configuration, force bool) *Dispatcher {
+	isVC := conf.Session.IsVC()
+	e := &Dispatcher{
+		session: conf.Session,
+		ctx:     conf.Context,
+		isVC:    isVC,
+		force:   force,
+	}
+	e.initDiagnosticLogs(conf)
+	e.networks = make(map[string]object.NetworkReference)
+	e.nics = make(map[string]string)
+	return e
+}
+
+// Get the current log header LineEnd of the hostd/vpxd logs.
+// With this we avoid collecting log file data that existed prior to install.
+func (d *Dispatcher) initDiagnosticLogs(conf *configuration.Configuration) {
+	if d.isVC {
+		diagnosticLogs[d.session.ServiceContent.About.InstanceUuid] =
+			&diagnosticLog{"vpxd:vpxd.log", "vpxd.log", 0, nil, true}
+	}
+
+	if d.session.Host == nil {
+		// vCenter w/ auto DRS.
+		// Set collect=false here as we do not want to collect all hosts logs,
+		// just the hostd log where the VM is placed.
+		for _, host := range conf.Hosts {
+			diagnosticLogs[host.Reference().Value] =
+				&diagnosticLog{"hostd", "hostd.log", 0, host, false}
+		}
+	} else {
+		// vCenter w/ manual DRS or standalone ESXi
+		var host *object.HostSystem
+		host = d.session.Host
+
+		diagnosticLogs[d.session.Host.Reference().Value] =
+			&diagnosticLog{"hostd", "hostd.log", 0, host, true}
+	}
+
+	m := diagnostic.NewDiagnosticManager(d.session)
+
+	for k, l := range diagnosticLogs {
+		// get LineEnd without any LineText
+		h, err := m.BrowseLog(d.ctx, l.host, l.key, math.MaxInt32, 0)
+
+		if err != nil {
+			log.Warnf("Disabling %s %s collection (%s)", k, l.name, err)
+			diagnosticLogs[k] = nil
+			continue
+		}
+
+		l.start = h.LineEnd
+	}
+}
+
+func (d *Dispatcher) Dispatch(conf *configuration.Configuration, timeout time.Duration) error {
+	var err error
+	if d.vchPool, err = d.createResourcePool(conf); err != nil && !d.force {
+		return errors.Errorf("Creating resource pool failed with %s. Exiting...", err)
+	}
+
+	if err = d.createNetworkIfNotExist(conf); err != nil && !d.force {
```

Comment:
This is specific to the bridge network so let's call this createBridgeNetwork...